StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1500CurlyBracketsForMultiLineStatementsMustNotShareLine.cs

```diff
@@ -73,7 +76,82 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            CheckCurlyBracketToken(context, ((NamespaceDeclarationSyntax)context.Node).CloseBraceToken);
+        }
+
+        private void HandleClassDeclarationSyntax(SyntaxNodeAnalysisContext context)
+        {
+            CheckCurlyBracketToken(context, ((ClassDeclarationSyntax)context.Node).OpenBraceToken);
+            CheckCurlyBracketToken(context, ((ClassDeclarationSyntax)context.Node).CloseBraceToken);
+        }
+
+        private void HandleEnumDeclarationSyntax(SyntaxNodeAnalysisContext context)
+        {
+            CheckCurlyBracketToken(context, ((EnumDeclarationSyntax)context.Node).OpenBraceToken);
+            CheckCurlyBracketToken(context, ((EnumDeclarationSyntax)context.Node).CloseBraceToken);
+        }
+
+        private void HandleInterfaceDeclarationSyntax(SyntaxNodeAnalysisContext context)
+        {
+            CheckCurlyBracketToken(context, ((InterfaceDeclarationSyntax)context.Node).OpenBraceToken);
+            CheckCurlyBracketToken(context, ((InterfaceDeclarationSyntax)context.Node).CloseBraceToken);
+        }
+
+        private void HandleStructDeclarationSyntax(SyntaxNodeAnalysisContext context)
+        {
+            CheckCurlyBracketToken(context, ((StructDeclarationSyntax)context.Node).OpenBraceToken);
+            CheckCurlyBracketToken(context, ((StructDeclarationSyntax)context.Node).CloseBraceToken);
+        }
+
+        private void HandleAccessorListSyntax(SyntaxNodeAnalysisContext context)
+        {
+            CheckCurlyBracketToken(context, ((AccessorListSyntax)context.Node).OpenBraceToken);
+            CheckCurlyBracketToken(context, ((AccessorListSyntax)context.Node).CloseBraceToken);
+        }
+
+        private void HandleBlockSyntax(SyntaxNodeAnalysisContext context)
+        {
+            CheckCurlyBracketToken(context, ((BlockSyntax)context.Node).OpenBraceToken);
+            CheckCurlyBracketToken(context, ((BlockSyntax)context.Node).CloseBraceToken);
+        }
+
+        private void CheckCurlyBracketToken(SyntaxNodeAnalysisContext context, SyntaxToken token)
+        {
+            if (token.IsMissing)
+                return;
+
+            Location location = token.GetLocation();
+            int line = location.GetLineSpan().StartLinePosition.Line;
+
+            SyntaxToken previousToken = token.GetPreviousToken();
+            if (!previousToken.IsMissing)
+            {
+                if (previousToken.GetLocation().GetLineSpan().StartLinePosition.Line == line)
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, location));
+                    // no need to report more than one instance for this token
+                    return;
+                }
+            }
+
+            SyntaxToken nextToken = token.GetNextToken();
+            if (!nextToken.IsMissing)
+            {
+                if (nextToken.GetLocation().GetLineSpan().StartLinePosition.Line == line)
```

Comment:
I'm waiting for the code fix for this to be part of the project before I really worry about it (it'll be included when we switch to Alpha 5).